### DISCUSSION TITLE
Can I update the code on my older Hardware revision?

### DISCUSSION THREAD

**Discussion by reclaimerx10** [sanni/cartreader#908] [Need help]
I know this is probably a very noob ask but brain is having a lull today lol 

So I have the old Hardware revision 4 and was wondering if it is updateable to the newest version of code or is it even worth it? Looking at trying to reflash GBC carts from aliexpress if its possible? or is there a need to patch the rom for batteryless saves or something like when you do the same with the GBA carts?

**Comment by Ancyker** [comment]
This is really a few questions, but to answer this one:

> So I have the old Hardware revision 4 and was wondering if it is updateable to the newest version of code or is it even worth it?

Yes.

**Comment by sanni** [comment]
> Looking at trying to reflash GBC carts from aliexpress if its possible?

Currently there are the DIY MBC3 & MBC5 options for this kind of cartridge: https://www.jrodrigo.net/wiki/how-to-diy-a-gameboy-flash-cartridge-with-a-rom-adapter-board/
The HDR MBC30 option for this cart: https://github.com/HDR/MBC30-Flashcart
And the HDR GameBoy Cam for this: https://github.com/HDR/Gameboy-Camera-Flashcart

There is also a more general CFI option that could work with Aliexpress carts if they have a compatible flash protocol but I have not tried it yet myself, so no promises.
